contracts/ERC20MultiDelegate.sol

```diff
@@ -173,9 +191,63 @@ contract ERC20MultiDelegate is ERC1155, Ownable {
         require(token.transferFrom(proxyAddressFrom, msg.sender, amount));
     }
 
-    function setUri(string memory uri) external onlyOwner {
-        _setURI(uri);
-        emit MetadataURIUpdated(uri);
+    function setUri(string memory /*uri*/) external view onlyOwner {
+        revert NotImplemented();
+    }
+
+    function tokenURI(uint256 tokenId) public view returns (string memory) {
+        string memory addressString = Strings.toHexString(uint160(tokenId), 20);
+        string memory reversedName = string.concat(
+            this.trim(addressString, 2, bytes(addressString).length),
+            ".addr.reverse"
+        );
+        (bytes memory encodedReversedName, ) = reversedName.dnsEncodeName();
+
+        string memory resolvedName;
+        try metadataResolver.reverse(encodedReversedName) returns (
+            string memory _resolvedName,
+            address,
+            address,
+            address
+        ) {
+            resolvedName = _resolvedName;
+        } catch {
+            return "";
```

Comment:
A failure to get the primary name shouldn't result in all the metadata being empty. Perhaps just return the hex-encoded address instead?